src/Hazelcast.Net/Core/LeaseTime.cs

```diff
@@ -12,24 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#nullable enable
+
 using System;
-using System.Threading;
 
 namespace Hazelcast.Core
 {
     /// <summary>
-    /// Utilities for managing time-to-live.
+    /// Provides lease-time constants.
     /// </summary>
-    public static class LeaseTime
+    public static partial class LeaseTime
     {
         /// <summary>
-        /// A constants used to specify an infinite lease time.
+        /// The max lease-time (according to server configuration).
         /// </summary>
-        public static readonly TimeSpan InfiniteTimeSpan = Timeout.InfiniteTimeSpan;
+        public static TimeSpan MaxValue => TimeSpanExtensions.MinusOneMillisecond;
```

Comment:
Why do we have `MaxValue` here, but have `Infinite` for others?

Reply:
Because... on the server it seems that you cannot specify an infinite lease time - it is either a specified value, or the default value which is configured on the server, and trying to use a value greater than that one returns an error. So... `MaxValue` tries to convey the idea that you're getting the maximum allowed value, which is not infinite.